File: src/main/java/com/github/aarcangeli/ideaclangformat/ClangFormatReplacement.java
package com.github.aarcangeli.ideaclangformat;

import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlText;

public class ClangFormatReplacement {
  @JacksonXmlProperty(isAttribute = true)
  public int offset;

  @JacksonXmlProperty(isAttribute = true)
  public int length;

  @JacksonXmlText(false)
  public String value;
}

File: src/main/kotlin/com/github/aarcangeli/ideaclangformat/services/ClangFormatReplacement.kt
package com.github.aarcangeli.ideaclangformat.services

import javax.xml.bind.annotation.XmlAttribute
import javax.xml.bind.annotation.XmlValue

class ClangFormatReplacement {
  @XmlAttribute
  var offset = 0

  @XmlAttribute
  var length = 0

  @XmlValue
  var value: String? = null
}
